rpc/v8/transaction.go

```diff
@@ -364,29 +366,53 @@ func adaptBroadcastedTransaction(broadcastedTxn *BroadcastedTransaction,
 	return txn, declaredClass, paidFeeOnL1, nil
 }
 
-func adaptResourceBounds(rb map[core.Resource]core.ResourceBounds) map[Resource]ResourceBounds {
-	rpcResourceBounds := make(map[Resource]ResourceBounds)
-	for resource, bounds := range rb {
-		rpcResourceBounds[Resource(resource)] = ResourceBounds{
-			MaxAmount:       new(felt.Felt).SetUint64(bounds.MaxAmount),
-			MaxPricePerUnit: bounds.MaxPricePerUnit,
+func adaptResourceBounds(rb map[core.Resource]core.ResourceBounds) *TransactionResourceBounds {
+	if len(rb) == 0 {
+		return &TransactionResourceBounds{}
+	}
+
+	return &TransactionResourceBounds{
+		L1Gas:     getResourceBounds(rb, core.ResourceL1Gas),
+		L1DataGas: getResourceBounds(rb, core.ResourceL1DataGas),
+		L2Gas:     getResourceBounds(rb, core.ResourceL2Gas),
+	}
+}
+
+func getResourceBounds(rb map[core.Resource]core.ResourceBounds, resource core.Resource) *ResourceBounds {
+	if coreBounds, ok := rb[resource]; ok {
+		return &ResourceBounds{
+			MaxAmount:       new(felt.Felt).SetUint64(coreBounds.MaxAmount),
+			MaxPricePerUnit: coreBounds.MaxPricePerUnit,
 		}
 	}
-	return rpcResourceBounds
+	return nil
```

Comment:
Please return a value type as well. Add a bool flag `ok` in case the `nil` response is used for something. 

```suggestion
func getResourceBounds(rb map[core.Resource]core.ResourceBounds, resource core.Resource) (ResourceBounds, bool) {
	if coreBounds, ok := rb[resource]; ok {
		return ResourceBounds{
			MaxAmount:       new(felt.Felt).SetUint64(coreBounds.MaxAmount),
			MaxPricePerUnit: coreBounds.MaxPricePerUnit,
		}, true	}
	return ResourceBounds{}, false
```

It will be later used as:

```go
resourceBounds, ok := getResourceBounds(...)
if !ok {
   ... 
}
```

This function with the new definition makes 0 heap allocations meaning is more easy on the memory, even if it seems it is actually more because it is returning two values instead of one. Also, it matches in style with go code style